pkg/utils/secrets/generate.go

```diff
@@ -67,10 +78,22 @@ func GenerateClusterSecrets(ctx context.Context, k8sClusterClient kubernetes.Int
 					Name:      s.GetName(),
 					Namespace: namespace,
 				},
-				Type: secretType,
-				Data: obj.SecretData(),
 			}
-			err = k8sClusterClient.Client().Create(ctx, secret)
+
+			err = kutil.CreateOrUpdate(ctx, k8sClusterClient.Client(), secret, func() error {
```

Comment:
Can we move the adding of the label a separate step and to `pkg/botanist/secrets.go`? This location here is more meant to be a utility library to generate secrets - it should not contain any specifics.

Reply:
Yes, sure - thanks for pointing that out. I've changed it.